src/utils/utils-vega.js

```diff
@@ -193,3 +194,448 @@ export function createRasterLayerGetterSetter(
     return layerObj
   }
 }
+
+
+// Polygon and line svg on hovering
+
+// NOTE: Reqd until ST_Transform supported on projection columns
+function conv4326To900913(x, y) {
+  const transCoord = [0.0, 0.0]
+  transCoord[0] = x * 111319.49077777777778
+  transCoord[1] =
+    Math.log(Math.tan((90.0 + y) * 0.00872664625997)) * 6378136.99911215736947
+  return transCoord
+}
+
+class PolySvgFormatter {
+  /**
+   * Builds the bounds from the incoming poly data
+   * @param {AABox2d} out AABox2d to return
+   * @param {object} data Object with return data from getResultRowForPixel()
+   * @param {Number} width Width of the visualization div
+   * @param {Number} height Height of the visualization div
+   * @param {object} margin Margins of the visualization div
+   * @param {Function} xscale d3 scale in x dimension from world space to pixel space (i.e. mercatorx-to-pixel)
+   * @param {Function} yscale d3 scale in y dimension from world space to pixel space (i.e. mercatory-to-pixel)
+   */
+  getBounds(data, width, height, margins, xscale, yscale) {
+    throw new Error("This must be overridden")
+  }
+
+  /**
+   * Builds the svg path string to use with the d svg attr:
+   * https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d
+   * This function should be called after the getBounds().
+   * The t/s arguments are the transformations to properly place the points underneath
+   * a parent SVG group node. That node is what ultimately handles animations and such
+   * so we need to transform all the points into local space. t is the translation
+   * and s is the scale to transform the points from pixel space to model/object space.
+   * @param {string} out Returns the svg path string
+   * @param {Point2d} t Translation from world to object space.
+   * @param {Number} s Scale from world to object space.
+   */
+  getSvgPath(t, s) {
+    throw new Error("This must be overridden")
+  }
+}
+
+class LegacySvgFormatter extends PolySvgFormatter {
```

Comment:
This legacy formatter is for polys only, so let's rename this class LegacyPolySvgFormatter.
It's for backwards compatibility, but could be removed if > 4.x releases of immerse cannot hit a 3.x server.